about/src/main/java/com/wangdaye/about/model/AppObject.java

```diff
@@ -27,4 +28,19 @@ public AppObject(int id, int iconId, String text) {
     public int getType() {
         return type;
     }
+
+    @Override
+    public boolean areItemsTheSame(BaseAdapter.ViewModel newModel) {
+        return newModel instanceof AppObject && ((AppObject) newModel).iconId == iconId;
```

Comment:
I detect that this code is problematic. According to the [Correctness (CORRECTNESS)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#correctness-correctness), [SA: Self comparison of value with itself (SA_SELF_COMPARISON)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#sa-self-comparison-of-field-with-itself-sa-field-self-comparison).
This method compares a field with itself, and may indicate a typo or a logic error. Make sure that you are comparing the right things.